openwisp_controller/api/serializers.py

```diff
@@ -0,0 +1,137 @@
+import collections
+import json
+
+from rest_framework import serializers
+from swapper import load_model
+
+from openwisp_users.api.mixins import FilterSerializerByOrgOwned
+from openwisp_utils.api.serializers import ValidatedModelSerializer
+
+Template = load_model('config', 'Template')
+Vpn = load_model('config', 'Vpn')
+Device = load_model('config', 'Device')
+Config = load_model('config', 'Config')
+TemplateTag = load_model('config', 'TemplateTag')
+Organization = load_model('openwisp_users', 'Organization')
+
+
+class BaseMeta:
+    read_only_fields = ['created', 'modified']
+
+
+class TemplateSerializer(ValidatedModelSerializer):
+    config = serializers.JSONField()
+    tags = serializers.StringRelatedField(many=True, read_only=True)
+    default_values = serializers.JSONField()
+    organization = serializers.PrimaryKeyRelatedField(
+        allow_null=False, queryset=Organization.objects.all(), required=True
+    )
+
+    class Meta(BaseMeta):
+        model = Template
+        fields = [
+            'id',
+            'name',
+            'tags',
+            'organization',
+            'type',
+            'backend',
+            'default',
+            'required',
+            'default_values',
+            'config',
+            'created',
+            'modified',
+        ]
+
+
+class VpnSerializer(FilterSerializerByOrgOwned, ValidatedModelSerializer):
+    config = serializers.JSONField()
+
+    class Meta(BaseMeta):
+        model = Vpn
+        fields = '__all__'
```

Comment:
please write the list of fields again so they're in the same order that they're listed in the admin, alternatively you may get the list of fields from `VpnAdmin.fields`

Reply:
Done